### ISSUE TITLE
Error messages on using cluster diagnosis on win 10 with bash

### ISSUE THREAD

**Issue by Shinzu** [cilium/cluster-diagnosis#19]
Hello,

when i run cluster-diagnosis via bash ( provided by git-sdk 64bit ) i get following errors:

```
$ python2.7 cluster-diagnosis.zip sysdump                                                                                                                                                                                                                                                                                                                            [2/1511]
INFO collecting nodes overview ...
The filename, directory name, or volume label syntax is incorrect.
?[1;41mERROR?[1;0m Error: Command 'kubectl get nodes -o json > ./cilium-sysdump-20180603-110554/nodes-2018-06-03T10:05:54.316719.json' returned non-zero exit status 1. Could not co
llect nodes overview: nodes-2018-06-03T10:05:54.316719.json
INFO collecting pods overview ...
The filename, directory name, or volume label syntax is incorrect.
?[1;41mERROR?[1;0m Error: Command 'kubectl get pods -o json --all-namespaces > ./cilium-sysdump-20180603-110554/pods-2018-06-03T10:05:54.336207.json' returned non-zero exit status
1. Could not collect pods overview: pods-2018-06-03T10:05:54.336207.json
INFO collecting pods summary ...
The filename, directory name, or volume label syntax is incorrect.
?[1;41mERROR?[1;0m Error: Command 'kubectl get pods --all-namespaces -o wide > ./cilium-sysdump-20180603-110554/pods-2018-06-03T10:05:54.346287.txt' returned non-zero exit status 1
. Could not collect pods summary: pods-2018-06-03T10:05:54.346287.txt
INFO collecting cilium gops stats ...
The filename, directory name, or volume label syntax is incorrect.
?[1;41mERROR?[1;0m Error: Command 'kubectl exec -it -n kube-system cilium-74pjf -- /bin/gops stats 1 > ./cilium-sysdump-20180603-110554/cilium-74pjf-2018-06-03T10:05:54.507567-stat
s.txt' returned non-zero exit status 1. Could not collect gops stats: cilium-74pjf-2018-06-03T10:05:54.507567-stats.txt
The filename, directory name, or volume label syntax is incorrect.
?[1;41mERROR?[1;0m Error: Command 'kubectl exec -it -n kube-system cilium-786lw -- /bin/gops stats 1 > ./cilium-sysdump-20180603-110554/cilium-786lw-2018-06-03T10:05:54.516975-stat
s.txt' returned non-zero exit status 1. Could not collect gops stats: cilium-786lw-2018-06-03T10:05:54.516975-stats.txt
The filename, directory name, or volume label syntax is incorrect.
?[1;41mERROR?[1;0m Error: Command 'kubectl exec -it -n kube-system cilium-t4l7n -- /bin/gops stats 1 > ./cilium-sysdump-20180603-110554/cilium-t4l7n-2018-06-03T10:05:54.525711-stat
s.txt' returned non-zero exit status 1. Could not collect gops stats: cilium-t4l7n-2018-06-03T10:05:54.525711-stats.txt
The filename, directory name, or volume label syntax is incorrect.
?[1;41mERROR?[1;0m Error: Command 'kubectl exec -it -n kube-system cilium-zr4c2 -- /bin/gops stats 1 > ./cilium-sysdump-20180603-110554/cilium-zr4c2-2018-06-03T10:05:54.536463-stat
s.txt' returned non-zero exit status 1. Could not collect gops stats: cilium-zr4c2-2018-06-03T10:05:54.536463-stats.txt
The filename, directory name, or volume label syntax is incorrect.
?[1;41mERROR?[1;0m Error: Command 'kubectl exec -it -n kube-system cilium-74pjf -- /bin/gops memstats 1 > ./cilium-sysdump-20180603-110554/cilium-74pjf-2018-06-03T10:05:54.688069-m
emstats.txt' returned non-zero exit status 1. Could not collect gops memstats: cilium-74pjf-2018-06-03T10:05:54.688069-memstats.txt
The filename, directory name, or volume label syntax is incorrect.
?[1;41mERROR?[1;0m Error: Command 'kubectl exec -it -n kube-system cilium-786lw -- /bin/gops memstats 1 > ./cilium-sysdump-20180603-110554/cilium-786lw-2018-06-03T10:05:54.697029-m
emstats.txt' returned non-zero exit status 1. Could not collect gops memstats: cilium-786lw-2018-06-03T10:05:54.697029-memstats.txt
The filename, directory name, or volume label syntax is incorrect.
?[1;41mERROR?[1;0m Error: Command 'kubectl exec -it -n kube-system cilium-t4l7n -- /bin/gops memstats 1 > ./cilium-sysdump-20180603-110554/cilium-t4l7n-2018-06-03T10:05:54.710469-m
emstats.txt' returned non-zero exit status 1. Could not collect gops memstats: cilium-t4l7n-2018-06-03T10:05:54.710469-memstats.txt
The filename, directory name, or volume label syntax is incorrect.
?[1;41mERROR?[1;0m Error: Command 'kubectl exec -it -n kube-system cilium-zr4c2 -- /bin/gops memstats 1 > ./cilium-sysdump-20180603-110554/cilium-zr4c2-2018-06-03T10:05:54.721989-m
emstats.txt' returned non-zero exit status 1. Could not collect gops memstats: cilium-zr4c2-2018-06-03T10:05:54.721989-memstats.txt
The filename, directory name, or volume label syntax is incorrect.
?[1;41mERROR?[1;0m Error: Command 'kubectl exec -it -n kube-system cilium-74pjf -- /bin/gops stack 1 > ./cilium-sysdump-20180603-110554/cilium-74pjf-2018-06-03T10:05:54.874949-stac
k.txt' returned non-zero exit status 1. Could not collect gops stack: cilium-74pjf-2018-06-03T10:05:54.874949-stack.txt
The filename, directory name, or volume label syntax is incorrect.
?[1;41mERROR?[1;0m Error: Command 'kubectl exec -it -n kube-system cilium-786lw -- /bin/gops stack 1 > ./cilium-sysdump-20180603-110554/cilium-786lw-2018-06-03T10:05:54.885189-stac
k.txt' returned non-zero exit status 1. Could not collect gops stack: cilium-786lw-2018-06-03T10:05:54.885189-stack.txt
The filename, directory name, or volume label syntax is incorrect.
?[1;41mERROR?[1;0m Error: Command 'kubectl exec -it -n kube-system cilium-t4l7n -- /bin/gops stack 1 > ./cilium-sysdump-20180603-110554/cilium-t4l7n-2018-06-03T10:05:54.894149-stac
k.txt' returned non-zero exit status 1. Could not collect gops stack: cilium-t4l7n-2018-06-03T10:05:54.894149-stack.txt
The filename, directory name, or volume label syntax is incorrect.
?[1;41mERROR?[1;0m Error: Command 'kubectl exec -it -n kube-system cilium-zr4c2 -- /bin/gops stack 1 > ./cilium-sysdump-20180603-110554/cilium-zr4c2-2018-06-03T10:05:54.903109-stac
k.txt' returned non-zero exit status 1. Could not collect gops stack: cilium-zr4c2-2018-06-03T10:05:54.903109-stack.txt
INFO collecting cilium network policy ...
The filename, directory name, or volume label syntax is incorrect.
?[1;41mERROR?[1;0m Error: Command 'kubectl get cnp -o yaml --all-namespaces > ./cilium-sysdump-20180603-110554/cnp-2018-06-03T10:05:54.913349.yaml' returned non-zero exit status 1.
 Could not collect cilium network policy: cnp-2018-06-03T10:05:54.913349.yaml
INFO collecting cilium-bugtool output ...
tar: Removing leading `/' from member names
INFO collected cilium-bugtool output: bugtool-cilium-74pjf-20180603-110555.tar
tar: Removing leading `/' from member names
INFO collected cilium-bugtool output: bugtool-cilium-786lw-20180603-110559.tar
tar: Removing leading `/' from member names
INFO collected cilium-bugtool output: bugtool-cilium-t4l7n-20180603-110602.tar
tar: Removing leading `/' from member names
INFO collected cilium-bugtool output: bugtool-cilium-zr4c2-20180603-110659.tar
INFO collecting cilium logs ...
The filename, directory name, or volume label syntax is incorrect.
?[1;41mERROR?[1;0m Error: Command 'kubectl logs  --timestamps=true --since=12h --limit-bytes=268435456 -n kube-system cilium-74pjf > ./cilium-sysdump-20180603-110554/cilium-74pjf-2
018-06-03T10:07:05.134076.log' returned non-zero exit status 1. Could not collect log file: cilium-74pjf-2018-06-03T10:07:05.134076
The filename, directory name, or volume label syntax is incorrect.
DEBUG Debug: Command 'kubectl logs --previous --timestamps=true --since=12h --limit-bytes=268435456 -n kube-system cilium-74pjf > ./cilium-sysdump-20180603-110554/cilium-74pjf-2018
-06-03T10:07:05.134076-previous.log' returned non-zero exit status 1. Could not collect previous log for 'cilium-74pjf': cilium-74pjf-2018-06-03T10:07:05.134076
The filename, directory name, or volume label syntax is incorrect.
?[1;41mERROR?[1;0m Error: Command 'kubectl logs  --timestamps=true --since=12h --limit-bytes=268435456 -n kube-system cilium-786lw > ./cilium-sysdump-20180603-110554/cilium-786lw-2
018-06-03T10:07:05.155403.log' returned non-zero exit status 1. Could not collect log file: cilium-786lw-2018-06-03T10:07:05.155403
The filename, directory name, or volume label syntax is incorrect.
DEBUG Debug: Command 'kubectl logs --previous --timestamps=true --since=12h --limit-bytes=268435456 -n kube-system cilium-786lw > ./cilium-sysdump-20180603-110554/cilium-786lw-2018
-06-03T10:07:05.155403-previous.log' returned non-zero exit status 1. Could not collect previous log for 'cilium-786lw': cilium-786lw-2018-06-03T10:07:05.155403
The filename, directory name, or volume label syntax is incorrect.
?[1;41mERROR?[1;0m Error: Command 'kubectl logs  --timestamps=true --since=12h --limit-bytes=268435456 -n kube-system cilium-t4l7n > ./cilium-sysdump-20180603-110554/cilium-t4l7n-2
018-06-03T10:07:05.176731.log' returned non-zero exit status 1. Could not collect log file: cilium-t4l7n-2018-06-03T10:07:05.176731
The filename, directory name, or volume label syntax is incorrect.
DEBUG Debug: Command 'kubectl logs --previous --timestamps=true --since=12h --limit-bytes=268435456 -n kube-system cilium-t4l7n > ./cilium-sysdump-20180603-110554/cilium-t4l7n-2018
-06-03T10:07:05.176731-previous.log' returned non-zero exit status 1. Could not collect previous log for 'cilium-t4l7n': cilium-t4l7n-2018-06-03T10:07:05.176731
The filename, directory name, or volume label syntax is incorrect.
?[1;41mERROR?[1;0m Error: Command 'kubectl logs  --timestamps=true --since=12h --limit-bytes=268435456 -n kube-system cilium-zr4c2 > ./cilium-sysdump-20180603-110554/cilium-zr4c2-2
018-06-03T10:07:05.201532.log' returned non-zero exit status 1. Could not collect log file: cilium-zr4c2-2018-06-03T10:07:05.201532
The filename, directory name, or volume label syntax is incorrect.
DEBUG Debug: Command 'kubectl logs --previous --timestamps=true --since=12h --limit-bytes=268435456 -n kube-system cilium-zr4c2 > ./cilium-sysdump-20180603-110554/cilium-zr4c2-2018
-06-03T10:07:05.201532-previous.log' returned non-zero exit status 1. Could not collect previous log for 'cilium-zr4c2': cilium-zr4c2-2018-06-03T10:07:05.201532
INFO deleting directory: ./cilium-sysdump-20180603-110554
INFO the sysdump has been saved in the file ./cilium-sysdump-20180603-110554.zip.
```

seems  that it is just a path related issue due to the strange way this works under the git bash on windows

**Comment by ashwinp** [comment]
Fixed in cluster-diagnosis v0.11

**Comment by Shinzu** [comment]
looks good
```
shinz@quincha MINGW64 ~/git/minikube (master)
$ python2.7 cluster-diagnosis.zip sysdump
INFO collecting kubernetes version info ...
INFO collected kubernetes version info: k8s-version-info-20180603-184350.yaml
INFO collecting nodes overview ...
INFO collected nodes overview: nodes-20180603-184350.json
INFO collecting pods overview ...
INFO collected pods overview: pods-20180603-184350.json
INFO collecting pods summary ...
INFO collected pods summary: pods-20180603-184350.txt
INFO collecting services overview ...
INFO collected svc overview: services-20180603-184350.yaml
INFO collecting cilium gops stats ...
INFO collected gops stats file: cilium-74pjf-20180603-184351-stats.txt
INFO collected gops stats file: cilium-786lw-20180603-184351-stats.txt
INFO collected gops stats file: cilium-v5gxj-20180603-184351-stats.txt
INFO collected gops stats file: cilium-zr4c2-20180603-184351-stats.txt
INFO collected gops memstats file: cilium-74pjf-20180603-184351-memstats.txt
INFO collected gops memstats file: cilium-786lw-20180603-184352-memstats.txt
INFO collected gops memstats file: cilium-v5gxj-20180603-184352-memstats.txt
INFO collected gops memstats file: cilium-zr4c2-20180603-184352-memstats.txt
INFO collected gops stack file: cilium-74pjf-20180603-184352-stack.txt
INFO collected gops stack file: cilium-786lw-20180603-184353-stack.txt
INFO collected gops stack file: cilium-v5gxj-20180603-184353-stack.txt
INFO collected gops stack file: cilium-zr4c2-20180603-184353-stack.txt
INFO collecting cilium network policy ...
INFO collected cilium network policy: cnp-20180603-184353.yaml
INFO collecting cilium daemonset yaml ...
INFO collected cilium daemonset yaml file: cilium-ds-20180603-184353.yaml
INFO collecting cilium configmap yaml ...
INFO collected cilium configmap yaml file: cilium-configmap-20180603-184353.yaml
INFO collecting cilium-bugtool output ...
tar: Removing leading `/' from member names
INFO collected cilium-bugtool output: bugtool-cilium-74pjf-20180603-184354.tar
tar: Removing leading `/' from member names
INFO collected cilium-bugtool output: bugtool-cilium-786lw-20180603-184400.tar
command terminated with exit code 137
ERROR Error: Command 'kubectl exec -n kube-system -it cilium-v5gxj cilium-bugtool' returned non-zero exit status 137. Could not run cilium-bugtool on cilium-v5gxj
tar: Removing leading `/' from member names
INFO collected cilium-bugtool output: bugtool-cilium-zr4c2-20180603-184430.tar
INFO collecting cilium logs ...
INFO collected log file: cilium-74pjf-20180603-184436
Error from server (BadRequest): previous terminated container "cilium-agent" in pod "cilium-74pjf" not found
DEBUG Debug: Command 'kubectl logs --previous --timestamps=true --since=12h --limit-bytes=268435456 -n kube-system cilium-74pjf > ./cilium-sysdump-20180603-184350/cilium-74pjf-201806
03-184436-previous.log' returned non-zero exit status 1. Could not collect previous log for 'cilium-74pjf': cilium-74pjf-20180603-184436
INFO collected log file: cilium-786lw-20180603-184436
Error from server (BadRequest): previous terminated container "cilium-agent" in pod "cilium-786lw" not found
DEBUG Debug: Command 'kubectl logs --previous --timestamps=true --since=12h --limit-bytes=268435456 -n kube-system cilium-786lw > ./cilium-sysdump-20180603-184350/cilium-786lw-201806
03-184436-previous.log' returned non-zero exit status 1. Could not collect previous log for 'cilium-786lw': cilium-786lw-20180603-184436
INFO collected log file: cilium-v5gxj-20180603-184437
INFO collected log file: cilium-v5gxj-20180603-184437-previous
INFO collected log file: cilium-zr4c2-20180603-184437
Error from server (BadRequest): previous terminated container "cilium-agent" in pod "cilium-zr4c2" not found
DEBUG Debug: Command 'kubectl logs --previous --timestamps=true --since=12h --limit-bytes=268435456 -n kube-system cilium-zr4c2 > ./cilium-sysdump-20180603-184350/cilium-zr4c2-201806
03-184437-previous.log' returned non-zero exit status 1. Could not collect previous log for 'cilium-zr4c2': cilium-zr4c2-20180603-184437
INFO deleting directory: ./cilium-sysdump-20180603-184350
INFO the sysdump has been saved in the file ./cilium-sysdump-20180603-184350.zip.
```
